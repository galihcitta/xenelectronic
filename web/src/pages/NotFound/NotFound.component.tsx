import React from 'react';
import { PageNotFound, Status, Message } from './NotFound.styles';

export const NotFound: React.FC = () => (
  <PageNotFound>
    <Status>404</Status>
    <Message>Page not found</Message>
  </PageNotFound>
);
