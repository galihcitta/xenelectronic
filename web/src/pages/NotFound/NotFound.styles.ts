import styled from 'styled-components';

import { color, fontWeight, device } from '../../utils/styles';

export const PageNotFound = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 20;
  background: ${color.white};
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const Status = styled.h2`
  font-size: 6rem;
  line-height: 6rem;

  @media ${device.tablet} {
    font-size: 12rem;
    line-height: 12rem;
  }
`;

export const Message = styled.p`
  font-size: 1.5rem;
  font-weight: ${fontWeight.semiBold};
  margin-bottom: 1.8em;

  @media ${device.tablet} {
    font-size: 3rem;
  }
`;
