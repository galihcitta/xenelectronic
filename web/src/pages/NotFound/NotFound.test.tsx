import React from 'react';
import { render } from '@testing-library/react';
import { NotFound } from './NotFound.component';

describe('<NotFound />', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });
});
