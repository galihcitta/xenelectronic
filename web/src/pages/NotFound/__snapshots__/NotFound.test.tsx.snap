// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa lcEBIK"
  >
    <h2
      class="sc-bwzfXH cVKDdH"
    >
      404
    </h2>
    <p
      class="sc-htpNat dIrHFr"
    >
      Page not found
    </p>
  </div>
</DocumentFragment>
`;
